modules/_labs/side-panel/react/README.md

```diff
@@ -0,0 +1,131 @@
+* `panelId: string`: specifies the id of the panel, if `undefined` then this hook will generate one
+* `labelId: string` specifies the id of the label, if `undefined` then this hook will generate one
+
+and returns:
+* `expanded: boolean`: The state for the `SidePanel`'s `expanded` prop.
+* `setExpanded: (expanded?: boolean) => void`: A function that sets the `expanded` state.
+* `panelProps: object`: Contains the `expanded`, `id`, and `aria-labelledby` props for the `SidePanel` (or equivalent) component.
+* `labelProps: object`: Contains the `id` prop for the label.
+* `controlProps: object`: Contains the `aria-controls`, `aria-expanded`, `aria-labelledby`, and click handler for the side panel's control element.
+
+#### Usage
+```tsx
+const [expanded, setExpanded, panelProps, labelProps, controlProps] = useSidePanel({
+  initialExpanded: true,
+  panelId: 'side-panel-1',
+  labelId: 'side-panel-label-1'
+});
+
+<SidePanel {...panelProps}>
+  {/* Make sure the control is first focusable */}
+  <SidePanel.ToggleButton {...controlProps} />
+  <span hidden {...labelProps}>Example Panel</span>
+  {/* Rest of the children here */}
+</SidePanel>
+
+or
+
+<SidePanel {...panelProps}>
+  {/* If you want to use your own button */}
+  <CustomControl {...controlProps} />
+  {/* If you already have an element that can be an accessible name for the panel */}
+  <h3 {...labelProps}>Example Panel</h3>
+</SidePanel>
+```
+
+## Static Properties
+
+#### `ToggleButton`
+> `<SidePanel.ToggleButton>` is a control that is meant to toggle between `expanded = true` and `expanded = false` states. It must be used within the `SidePanel` component as a child. Use in conjunction with `useSidePanel`'s `controlProps`, otherwise it does not come with explicit `onClick` handlers.
+
+> **Important**: For accessibility purposes, it must be the first focusable element. We recommend that you keep it as the first child of `SidePanel`.
+
+## Component Props
+
+### Required
+
+> None
+
+### Optional
+
+#### `as: React.ElementType`
+> The element the side panel will render as.
+
+Default: `'section'`
+
+#### `collapsedWidth: number | string`
+> The width of the component (in `px` if it's a `number`) when it is collapsed.
+
+Default: `64`
+
+#### `expanded: boolean`
+> If true, sets the expanded state of the side panel
+
+Default: `false`
+
+#### `expandedWidth: number | string`
+> The width of the component (in `px` if it's a `number`) when it is expanded.
+
+Default: `320`
+
+#### `height: number | string`
+> The height of the component (in `px` if it's a `number`).
+
+Default: `'100%'`
+
+#### `origin: 'left' | 'right'`
+> Which side the side panel is meant to originate from.
+
+Default: `'left'`
+
+#### `onExpandedChange: (expanded?: boolean) => void`
+> The function called when the side panel's `expanded` state changes. States like `'collapsing'` and `'expanding'` are tracked in another callback: `onStateChange`
```

Comment:
No conventions here for the weird internal state transition callback so it'll be flexible if we decide to change it. Here's the explanations for the names:
* `onStateTransition` is stating that this cb fires on internal state transitions
* `onExpandedChange` is the general React convention `on<Value>Change` so to disambiguate, I wrote `onStateTransition` rather than `onStateChange`... because `state` isn't a value/prop that's changing on the component necessarily.